src/cli/cli.cpp

```diff
@@ -151,6 +156,16 @@ std::ostream& Command::output() {
    return std::cout;
 }
 
+std::ostream& Command::output_binary() {
+   if(m_output_stream) {
+      return *m_output_stream;
+   }
+#ifdef _WIN32
+   _setmode(_fileno(stdout), _O_BINARY);
+#endif
```

Comment:
Mhh, juicy global state. 😞

Assuming one mixes `output()` and `output_binary()` for whatever reasons, we might end up with subtle misuse:

```C++
auto binstream = output_binary(); // silently sets the mode on `std::cout` to `_O_BINARY`
auto txtstream = output(); // `std::cout` is still set to binary
```

... or:

```C++
auto txtstream = output();
auto binstream = output_binary(); // silently sets the mode on `std::cout` to `_O_BINARY` (affecting txtstream as well)
```

Let's at least document that and discourage the mixed usage.

Reply:
I agree that changing the global state is not nice. But I don't think it is a big problem in this case.

If one writes binary data from stdin to e.g. pipe it to somewhere, any additional "normal" test output would likely break the use case anyway.

Reply:
added some doc strings

Reply:
Thanks.